src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -1111,9 +1112,31 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
 
     void Schedule::addGroup(const RestartIO::RstGroup& rst_group, std::size_t timeStep) {
         auto udq_undefined = this->getUDQConfig(timeStep).params().undefinedValue();
-        const auto& sched_state = this->snapshots.back();
-        auto insert_index = sched_state.groups.size();
-        this->addGroup( Group(rst_group, insert_index, udq_undefined, this->m_static.m_unit_system) );
+
+        const auto insert_index = this->snapshots.back().groups.size();
+
+        auto new_group = Group(rst_group, insert_index, udq_undefined, this->m_static.m_unit_system);
+
+        if (rst_group.name != "FIELD") {
+            // Common case.  Add new group.
+            this->addGroup( std::move(new_group) );
+            return;
+        }
```

Comment:
As I see it this is two blocks; 1) normal !field block  2) Special case for field; good if you could use the space to signal this grouping a bit - i.e. something like:
```C++
// block 1
const auto insert_index = this->snapshots.back().groups.size();
auto new_group = Group(rst_group, insert_index, udq_undefined, this->m_static.m_unit_system);
if (rst_group.name != "FIELD") {
            // Common case.  Add new group.
            this->addGroup( std::move(new_group) );
            return;
}

// if we are here ...

// block 2
auto& field = ..
if (new_group.isProd(...
for (const auto& phase : ...
```

Reply:
> good if you could use the space to signal this grouping a bit

I've pushed an update to that effect.